Fix SimpleIndexer fit method to set inputCol and outputCol correctly

MadDataScience opened the pull request:
This fixes #89. What was happening before is that after SimpleIndexer was fit to a dataframe, the inputCol and outputCol params were not carrying through to the new object. This change fixes that. I also added a couple tests.

Credit to @conorbmurphy for finding this bug and helping me figure it out.

holdenk commented on src/main/scala/com/high-performance-spark-examples/ml/CustomPipeline.scala:
```diff
@@ -122,7 +122,10 @@ class SimpleIndexer(override val uid: String)
       .collect()
-    new SimpleIndexerModel(uid, words)
+    val model = new SimpleIndexerModel(uid, words)
+    model.set(inputCol, $(inputCol))
```
Thanks for catching this! The proper fix is likely to use the `copyValues` function from the Params trait.

MadDataScience replied:
That was my first thought as well. (Actually, I tried using `copy`, not `copyValues`, but I think it would have the same problem.) Both `copy` and `copyValues` seem to need `extra` which is of type `ParamMap` but the `paramMap` property is private and `extractParamMap` just returns an empty map.

holdenk replied:
So `copy` is a bit different -- in `copyValues` the `extra` is an optional paramater (it has the default value of `ParamMap.empty`. If that doesn't work let me know and we can open a JIRA to fix the API on the Spark side for that.

holdenk commented:
Thanks for catching this, and especially thanks for adding a test. I think we can probably use the `copyValues` function and it might be a bit clearer (and easier to handle for people's own custom models with more parameters).

VoluntasTua commented:
I stand corrected. `copyValues` did the trick. Unfortunately, IntelliJ puked all over my commit. I hope you don't mind I added .idea/ to .gitignore. (I didn't mean to include that in my commit, but maybe just as well.)

MadDataScience commented:
I stand corrected. `copyValues` did the trick. Unfortunately IntelliJ puked all over my repo. I think I cleaned it up. I hope you don't mind I added .idea/ to .gitignore. I didn't mean to include that in my commit but it might be just as well.
